src/lib/statistics/statistics_objects/abstract_histogram.cpp

```diff
@@ -0,0 +1,1035 @@
+        return _does_not_contain(PredicateCondition::NotEquals, variant_value);
+      }
+
+      // If the pattern starts with a MatchAll, we can only prune it if it matches all values.
+      if (value->front() == '%') {
+        return *value == "%";
+      }
+
+      /**
+       * We can also prune prefix searches iff the domain of values captured by the histogram is less than or equal to
+       * the domain of strings captured by a prefix pattern.
+       *
+       * Example:
+       * min: car
+       * max: crime
+       * predicate: col NOT LIKE 'c%'
+       *
+       * With the same argument we can also prune predicates in the form of 'c%foo',
+       * where foo can be any pattern itself.
+       * We only have to consider the pattern up to the first MatchAll character.
+       */
+      const auto match_all_index = value->find('%');
+      if (match_all_index != pmr_string::npos) {
+        const auto search_prefix = value->substr(0, match_all_index);
+        if (search_prefix == bin_minimum(BinID{0}).substr(0, search_prefix.length()) &&
+            search_prefix == bin_maximum(bin_count() - 1).substr(0, search_prefix.length())) {
+          return true;
+        }
+      }
+
+      return false;
+    }
+    default: {
+      auto value_in_domain = _domain.string_to_domain(*value);
+
+      std::optional<AllTypeVariant> value2_in_domain;
+      if (variant_value2) {
+        value2_in_domain = boost::get<pmr_string>(*variant_value2);
+      }
+
+      return _general_does_not_contain(predicate_condition, value_in_domain, value2_in_domain);
+    }
+  }
+}
+
+template <typename T>
+CardinalityAndDistinctCountEstimate AbstractHistogram<T>::estimate_cardinality_and_distinct_count(
+    const PredicateCondition predicate_condition, const AllTypeVariant& variant_value,
+    const std::optional<AllTypeVariant>& variant_value2) const {
+  auto value = static_variant_cast<T>(variant_value);
+  if (!value) {
+    return {static_cast<Cardinality>(total_count()), static_cast<float>(total_distinct_count())};
+  }
+
+  if constexpr (std::is_same_v<
+                    T,
+                    pmr_string>) {  // NOLINT clang-tidy is crazy and sees a "potentially unintended semicolon" here...
+    value = _domain.string_to_domain(*value);
+  }
+
+  if (_does_not_contain(predicate_condition, variant_value, variant_value2)) {
+    return {Cardinality{0}, 0.0f};
+  }
+
+  switch (predicate_condition) {
+    case PredicateCondition::Equals: {
+      const auto bin_id = _bin_for_value(*value);
+      const auto bin_distinct_count = this->bin_distinct_count(bin_id);
+
+      if (bin_distinct_count == 0) {
+        return {Cardinality{0.0f}, 0.0f};
+      } else {
+        const auto cardinality = Cardinality{bin_height(bin_id) / bin_distinct_count};
+
+        if (bin_distinct_count == 1) {
+          return {cardinality, 1.0f};
+        } else {
+          return {cardinality, std::min(bin_distinct_count, HistogramCountType{1.0f})};
```

Comment:
min? When would we have `bin_distinct_count < 1`?

Reply:
histograms are statistical estimation. histograms resulting from previous estimations of predicates can easily have bin_heights/distinct counts < 1. We still want these numbers as they are, and not rounded up to 1, or something. This is so widespread that doc'ing this won't help. It is an intrinsic properties of statistics objects that they are estimations.